test/unit/org/apache/cassandra/auth/AuthCacheInvalidationTest.java

```diff
@@ -63,146 +83,161 @@ public void prepare() throws Throwable
         executeNet("DROP KEYSPACE IF EXISTS " + ks);
 
         executeNet("CREATE USER " + user1.getRoleName() + " WITH PASSWORD '" + userPw + '\'');
-        executeNet("CREATE ROLE " + role1.getRoleName());
-        executeNet("CREATE ROLE " + role2.getRoleName());
+        createRole(role1);
+        createRole(role2);
         executeNet("GRANT " + role1.getRoleName() + " TO " + user1.getRoleName());
         executeNet("GRANT " + role2.getRoleName() + " TO " + role1.getRoleName());
 
         execute("CREATE KEYSPACE " + ks + " WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 1}");
         execute("CREATE TABLE " + ks + '.' + tab + " (id int PRIMARY KEY)");
         execute("CREATE TABLE " + ks + '.' + tab2 + " (id int PRIMARY KEY)");
 
-        executeNet("GRANT SELECT, UPDATE ON TABLE " + ks + '.' + tab + " TO " + role1.getRoleName());
-        executeNet("GRANT UPDATE ON KEYSPACE " + ks + " TO " + role2.getRoleName());
+        executeNet("GRANT SELECT ON TABLE " + ks + '.' + tab + " TO " + role1.getRoleName());
+        executeNet("GRANT MODIFY ON TABLE " + ks + '.' + tab + " TO " + role1.getRoleName());
+        executeNet("GRANT MODIFY ON KEYSPACE " + ks + " TO " + role2.getRoleName());
 
-        DatabaseDescriptor.getAuthManager().invalidateCaches();
+        Roles.clearCache();
+        AuthenticatedUser.clearCache();
+    }
+
+    private void createRole(RoleResource role) throws Throwable
+    {
+        executeNet(format("CREATE ROLE %s WITH LOGIN = true AND password = '%s'", role.getRoleName(), userPw));
```

Comment:
Unlike DSE, in C* roles have to be themselves able to login and run a query for the permissions cache to contain entries for them.